lib/mongo/error/operation_failure.rb

```diff
@@ -81,12 +89,21 @@ def write_retryable?
       # @example Create the error object
       #   OperationFailure.new(message, result)
       #
-      # param [ String ] message The error message.
-      # param [ Operation::Result ] result The result object.
+      # @example Create the error object with a code and a code name
+      #   OperationFailure.new(message, result, :code => code, :code_name => code_name)
+      #
+      # @param [ String ] message The error message.
+      # @param [ Operation::Result ] result The result object.
+      # @param [ Hash ] options Additional parameters
+      #
+      # @option options [ Fixnum ] :code Error code
+      # @option options [ String ] :code_name Error code name
       #
-      # @since 2.5.0
-      def initialize(message = nil, result = nil)
+      # @since 2.5.0, options added in 2.6.0
```

Comment:
Out of curiosity, is this standard syntax? I'm wondering if it makes more sense to just version the options were added on line 97.

Reply:
That's for Javadocs though; we use [YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md#since). I wasn't sure if there were restrictions on `since` tags in the way that it formatted them, but from the link I gave, it doesn't appear so. I'm on the fence about putting the detail about the options on the line where the options are described versus on the `since` tag, so I'll defer to whatever you think is preferable.